### ISSUE TITLE
Integrate new .gitignore system

### ISSUE THREAD

**Issue by brynblack** [MatrixAI/nixpkgs-matrix#3]
### Specification

There exists a more advanced `.gitignore` system called [hercules-ci](https://github.com/hercules-ci/gitignore.nix). Our current `nix-gitignore.gitignoreSource` still keeps `.git` in the produced output.

```
nix-gitignore.gitignoreSource [] ./.
```

Still has `.git`.

But you need to do:

```
nix-gitignore.gitignoreSource [".git"] ./.
```

Actually ignores it.

Now I didn't see this before, not sure if this is a regression.

However the hercules-ci gitignore is apparently the more advanced one.

Basically get this done: https://github.com/hercules-ci/gitignore.nix/issues/6

### Tasks

1. ...
2. ...
3. ...

**Comment by aryanjassal** [comment]
According to [NixOS Wiki](https://nixos.wiki/wiki/flakes), for flakes in git repos, only files in the working tree will be copied to the store. Given that we are migrating our repos from using `shell.nix` to use `flake.nix`, this issue might be redundant.

**Comment by CMCDragonkai** [comment]
No, we still need to ignore more files then what is left in the working tree.

So it may be that we don't have to ignore `.git`, but we still have to ignore all other files. For example:

```
  src = nix-gitignore.gitignoreSource [
    # The `.git` itself should be ignored
    ".git"
    # Non-build files
    "/nodemon.json"
    # Hidden files
    "/.*"
    # Nix files
    "/*.nix"
    # Benchmarks
    "/benches"
    # Docs
    "/docs"
    # Tests
    "/tests"
    "/jest.config.js"
  ] ./.;
```

So this issue isn't redundant.

**Comment by tegefaulkes** [comment]
Getting this issue needs to be the focus for now @aryanj. Then go back to [ENG-358](https://linear.app/matrix-ai/issue/ENG-358/implement-secrets-ls-command). In the mean time I'll have a look over that and resolve the blocker you have for it.

**Comment by CMCDragonkai** [comment]
In the future, whoever is assigned to the issue should also write up a set of initial tasks - it doesn't have to be accurate, the accurate the tasks should go to the PR.
